### DISCUSSION TITLE
Boosting terms when using BM25 batch search

### DISCUSSION THREAD

**Discussion by Fatima-Haouari** [castorini/pyserini#1254] [Q&A]
Hi,
Could you please advise if there is way to boost some terms when using the bm25 batch search. I found only the [query builder example](https://github.com/castorini/pyserini/blob/master/docs/usage-querybuilder.md) where  boosting terms is possible which I believe it is not the same.

**Comment by GioiaZheng** [comment]
Right — `batch_search()` only takes plain query strings, so the query-builder boosts don't reach it (and a `term^2` in the string gets analyzed away as bag-of-words).

If you need per-term boosting, build the query yourself and call `search()` per query instead of `batch_search()`:

```python
from pyserini.search.lucene import querybuilder

should = querybuilder.JBooleanClauseOccur['should'].value
bq = querybuilder.get_boolean_query_builder()
for term, w in [('climate', 2.0), ('change', 1.0)]:
    tq = querybuilder.get_term_query(term)
    boosted = querybuilder.get_boost_query(tq, w)
    bq.add(boosted, should)
hits = searcher.search(bq.build())
```

You lose the built-in threading, but you can run your own thread pool over the queries to get it back. If you only need a rough up-weight and want to keep `batch_search()`, repeating a term in the query string raises its tf — blunt, but it works without leaving batch mode.
